.github/actions/BuildCirct/action.yml

```diff
@@ -4,28 +4,27 @@ description: "Builds CIRCT, which is used for the HLS backend"
 runs:
   using: "composite"
   steps:
+    - name: "Install LLVM, Clang, MLIR, and Ninja"
```

Comment:
I moved this first as it is independent of the caching. 
This keeps the cache steps together, which should make things clearer.